api/src/main/java/jakarta/data/metamodel/constraint/LikeLiteralRecord.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel.constraint;
+
+import jakarta.data.metamodel.restrict.Operator;
+
+import java.util.Objects;
+
+record LikeLiteralRecord(String literal, boolean caseSensitive)
+        implements Like {
+
+    public LikeLiteralRecord {
+        Objects.requireNonNull(literal, "Literal must not be null");
+    }
+
+    public LikeLiteralRecord(String literal) {
+        this(literal, true);
+    }
```

Comment:
```suggestion
record LikeLiteralRecord(String literal, boolean caseSensitive, boolean escaped)
        implements Like {

    public LikeLiteralRecord {
        Objects.requireNonNull(literal, "Literal must not be null");
        if (escaped) {
            throw new IllegalArgumentException("Literal values must not be escaped");
        }
    }

    public LikeLiteralRecord(String literal) {
        this(literal, true, false);
    }

    LikeLiteralRecord(String literal, boolean caseSensitive) {
        this(literal, caseSensitive, false);
    }
```

Reply:
It definitely doesn't make sense to say that a literal string is escaped or not. What could that possibly mean?

Reply:
> It definitely doesn't make sense to say that a literal string is escaped or not. What could that possibly mean?

Agree. It makes no sense for a literal to be escaped. But `LikeLiteralRecord` is an implementation of a `Like` interface that in some cases will be an escaped thing (although never when it is a literal).  The point is that the implementor who receives a `Like` constraint needs to known whether or not the string is escaped.  Having said that, using a record component that is forced to always be false was a bad choice for how to get an `escaped()` method that always returns false for  `LikeLiteralRecord`.  I'll remove the suggestions for that and replace them with a method that just returns false.